### PR TITLE
Added new version of BatchNormInference

### PR DESCRIPTION
**Pull request by ilyachur**

### CODE REVIEW

File: ngraph/core/include/ngraph/op/batch_norm.hpp [hunk 1]
```diff
@@ -66,6 +65,44 @@ namespace ngraph
...
+                void set_eps_value(double epsilon) { m_epsilon = epsilon; }
+                std::shared_ptr<Node>
+                    clone_with_new_inputs(const OutputVector& new_args) const override;
+
+            private:
+                static constexpr size_t INPUT_DATA = 0;
+                static constexpr size_t INPUT_GAMMA = 1;
+                static constexpr size_t INPUT_BETA = 2;
+                static constexpr size_t INPUT_MEAN = 3;
+                static constexpr size_t INPUT_VARIANCE = 4;
```
Comment (lazarevevgeny) [inline ngraph/core/include/ngraph/op/batch_norm.hpp:102]: nit: May be move this to protected? So they can be used if a new version of the operation is introduced.

File: ngraph/core/src/op/batch_norm.cpp [hunk 2]
```diff
@@ -67,9 +67,60 @@ void op::BatchNormInference::validate_and_infer_types()
...
+{
+    visitor.on_attribute("epsilon", m_epsilon);
+    return true;
+}
+
+void op::v5::BatchNormInference::validate_and_infer_types()
+{
+    element::Type result_et;
+    PartialShape result_batch_shape;
+    PartialShape result_channel_shape; // unused here
```
Comment (lazarevevgeny) [inline ngraph/core/src/op/batch_norm.cpp:101]: Remove?

File: inference-engine/src/transformations/src/transformations/op_conversions/batch_norm_decomposition.cpp [hunk 3]
```diff
@@ -43,39 +46,107 @@ ngraph::pass::BatchNormDecomposition::BatchNormDecomposition() {
...
+    ngraph::graph_rewrite_callback callback = [this, input, gamma, beta, mean, var](ngraph::pattern::Matcher &m) {
+        auto pattern_map = m.get_pattern_map();
+
+        auto m_input = pattern_map[input];
+        auto m_gamma = pattern_map[gamma];
+        auto m_beta = pattern_map[beta];
+        auto m_mean = pattern_map[mean];
+        auto m_var = pattern_map[var];
+
+        // TODO: check that all input shapes are static
```
Comment (lazarevevgeny) [inline inference-engine/src/transformations/src/transformations/op_conversions/batch_norm_decomposition.cpp:109]: Please, resolve the TODO

**Review by lazarevevgeny — APPROVED** [review]

File: ngraph/core/include/ngraph/op/batch_norm.hpp (hunk 1 above)
Comment (ilyachur) [reply ngraph/core/include/ngraph/op/batch_norm.hpp:102]: We can do these fields to protected section if we face with situation, which requires it. (it will be backward compatible changes).

File: inference-engine/src/transformations/src/transformations/op_conversions/batch_norm_decomposition.cpp (hunk 3 above)
Comment (GlebKazantaev) [reply inference-engine/src/transformations/src/transformations/op_conversions/batch_norm_decomposition.cpp:109]: Will be resolved as a part of https://github.com/openvinotoolkit/openvino/pull/2735

File: ngraph/core/include/ngraph/op/batch_norm.hpp (hunk 1 above)
Comment (GlebKazantaev) [reply ngraph/core/include/ngraph/op/batch_norm.hpp:102]: BTW why do we need this variables if we have clear order of inputs in ctor?

File: ngraph/test/type_prop/batch_norm.cpp [hunk 4]
```diff
@@ -69,7 +70,8 @@ TEST(type_prop, batch_norm_inference_partial_input_rank_static_dynamic_ok)
     auto mean = make_shared<op::Parameter>(mean_et, mean_shape);
     auto variance = make_shared<op::Parameter>(variance_et, variance_shape);
 
-    auto bn = make_shared<op::BatchNormInference>(data_batch, gamma, beta, mean, variance, epsilon);
+    auto bn =
+        make_shared<op::v5::BatchNormInference>(data_batch, gamma, beta, mean, variance, epsilon);
```
Comment (GlebKazantaev) [inline ngraph/test/type_prop/batch_norm.cpp:74]: Why we remove tests for v0 operation?

**Review by GlebKazantaev — CHANGES_REQUESTED** [review]
@lazarevevgeny @echuraev do we need to update ngrpah python API?

File: ngraph/core/include/ngraph/op/batch_norm.hpp (hunk 1 above)
Comment (ilyachur) [reply ngraph/core/include/ngraph/op/batch_norm.hpp:102]: We also use it in validate and infer types. Also it allowed to use the code from opset1 BatchNormInference without any changes.

**Comment by ilyachur** [comment]
> @lazarevevgeny @echuraev do we need to update ngrpah python API?

Did you ask me or my brother? 😃 
Yes I think we also need to update nGraph Python API. Will do it. Thank you

File: ngraph/test/type_prop/batch_norm.cpp (hunk 4 above)
Comment (ilyachur) [reply ngraph/test/type_prop/batch_norm.cpp:74]: Agreed, will add new tests for v5 BatchNormInference.

Comment (ilyachur) [reply ngraph/test/type_prop/batch_norm.cpp:74]: Fixed

**Comment by ilyachur** [comment]
All comments were fixed

File: ngraph/core/src/op/batch_norm.cpp (hunk 2 above)
Comment (ilyachur) [reply ngraph/core/src/op/batch_norm.cpp:101]: I cannot remove it. It used in the method infer_batch_norm_forward.

**Review by lazarevevgeny — APPROVED** [review]
Approved from the op definition and spec perspective.

**Comment by lazarevevgeny** [comment]
@ilyachur, please, fix the test.

**Comment by ilyachur** [comment]
> @ilyachur, please, fix the test.

Sure, I am working on it.

**Review by GlebKazantaev — APPROVED** [review]
